jcapi-systems.go

```diff
@@ -0,0 +1,334 @@
+	Id                             string  `json:id`
+	LastContact                    string  `json:lastContact`
+	RemoteIP                       string  `json:remoteIP`
+	Active                         bool    `json:active`
+	SshRootEnabled                 bool    `json:sshRootEnabled`
+	AmazonInstanceID               string  `json:amazonInstanceID,omitempty`
+	SshPassEnabled                 bool    `json:sshPassEnabled`
+	Version                        string  `json:version`
+	AgentVersion                   string  `json:agentVersion`
+	AllowPublicKeyAuth             bool    `json:allowPublicKeyAuthentication`
+	Organization                   string  `json:organization`
+	Created                        string  `json:created`
+	Arch                           string  `json:arch`
+	SystemTimezone                 float64 `json:systemTimeZone`
+	AllowSshPasswordAuthentication bool    `json:allowSshPasswordAuthentication`
+	DisplayName                    string  `json:displayName`
+	ModifySSHDConfig               bool    `json:modifySSHDConfig`
+	AllowMultiFactorAuthentication bool    `json:allowMultiFactorAuthentication`
+	Hostname                       string  `json:hostname`
+
+	TagList               []string `json:"tags"`
+	Patches               []string `json:patches`
+	SshParamList          []string `json:sshParams`
+	PatchAlarmList        []string `json:patchAlarms`
+	NetworkInterfaceList  []string `json:networkInterfaces`
+	ConnectionHistoryList []string `json:connectionHistory`
+
+	Tags              []JCTag
+	SshdParams        []JCSSHDParam
+	NetworkInterfaces []JCNetworkInterface
+}
+
+type JCSSHDParam struct {
+	Name  string `json:name`
+	Value string `json:value`
+}
+
+type JCNetworkInterface struct {
+	Name     string `json:name`
+	Internal bool   `json:internal`
+	Family   string `json:family`
+	Address  string `json:address`
+}
+
+func SystemsToString(systems []JCSystem) string {
+	returnVal := ""
+
+	for _, system := range systems {
+		returnVal += system.ToString()
+	}
+
+	return returnVal
+}
+
+func (jcsystem JCSystem) ToString() string {
+	//
+	// WARNING: Never output password via this method, it could be logged in clear text
+	//
+	//	returnVal := fmt.Sprintf("JCUSER: Id=[%s] - UserName=[%s] - FName/LName=[%s/%s] - Email=[%s] - sudo=[%t] - Uid=%s - Gid=%s - enableManagedUid=%t\n",
+	//		jcuser.Id, jcuser.UserName, jcuser.FirstName, jcuser.LastName,
+	//		jcuser.Email, jcuser.Sudo, jcuser.Uid, jcuser.Gid, jcuser.EnableManagedUid)
+	//
+	//	returnVal += fmt.Sprintf("JCUSER: ExternallyManaged=[%t] - ExternalDN=[%s] - ExternalSourceType=[%s]\n",
+	//		jcuser.ExternallyManaged, jcuser.ExternalDN, jcuser.ExternalSourceType)
+	//
+	//	for _, tag := range jcuser.Tags {
+	//		returnVal += fmt.Sprintf("\t%s\n", tag.ToString())
+	//	}
+	returnVal := "I'm a return val"
+	return returnVal
+}
+
+func getJCSSHDParamFieldsFromInterface(fields map[string]interface{}, params *JCSSHDParam) {
+	params.Name = fields["name"].(string)
```

Comment:
I don't believe that params *JCSSHDParam will allocate memory here, will it? I think it's just declaring a pointer, and so this could end up throwing a null pointer exception, I believe.

Probably add:

params = new(JCSSHDParam)